tests/e2e/experimental_throttle_test.go

```diff
@@ -0,0 +1,165 @@
+// Copyright 2016 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"go.etcd.io/etcd/tests/v3/framework/e2e"
+)
+
+const ENV_TEST_QMON = "ETCD_TEST_EXP_QMON"
+const ENV_TEST_QMON_VALUE = "yes"
+
+const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+func randStringBytes(n int) string {
+	b := make([]byte, n)
+	for i := range b {
+		b[i] = letterBytes[rand.Intn(len(letterBytes))]
+	}
+	return string(b)
+}
+
+func TestCtlV3RangeReqStorm(t *testing.T) {
+	if os.Getenv(ENV_TEST_QMON) != ENV_TEST_QMON_VALUE {
+		return
+	}
+
+	cfg := e2e.EtcdProcessClusterConfig{
+		ClusterSize:             1,
+		InitialToken:            "new",
+		EnableBandwidthThrottle: true,
+	}
+	testCtl(t, rangeReqTest, withCfg(cfg), withDialTimeout(5*time.Minute))
+}
+
+func rangeReq(cx ctlCtx, kvs []kv) {
+
+	tests := []struct {
+		args []string
+
+		wkv []kv
+	}{
+		{[]string{"key", "--prefix"}, kvs},
+	}
+
+	if err := ctlV3Get(cx, tests[0].args, tests[0].wkv...); err != nil {
+		if cx.dialTimeout > 0 && !isGRPCTimedout(err) && !isThrottleError(err) {
+			cx.t.Errorf("getTest ctlV3Get error (%v)", err)
```

Comment:
Can we show how many requests are throttled and measure if it is expected?

Reply:
concurrent bombardment of requests gets a push back outside of expected bandwidth limit. I will look into this one incrementally -  possibly add a test which builds up request pressure with some jitter instead of launching the requests concurrently